latest changes: cant compile under windows

### Version
latest version



### Operating system type + version
win10 64

### Behavior
An hour ago, I issued the normal GIT PULL, PERL BUILD.PL commands
In the build i get to this point:
t\22_config.t ................... ok

And it stays there for ever. I gave it like 15 minutes and just stayed there.
The git pull I did 1 week ago apart from the opengl problem, builded with no problems.

I just have issued a GIT PULL command, now there are errors reported in the build:

C:\dev\boost_1_59_0/boost/bind/bind.hpp:467:40: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
     template<class V> void accept(V & v) const
                                        ^
C:\dev\boost_1_59_0/boost/bind/bind.hpp:469:28: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
         base_type::accept(v);
                            ^
C:\dev\boost_1_59_0/boost/bind/bind.hpp:535:40: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
     template<class V> void accept(V & v) const
                                        ^
C:\dev\boost_1_59_0/boost/bind/bind.hpp:537:28: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
         base_type::accept(v);
                            ^
C:\dev\boost_1_59_0/boost/bind/bind.hpp:606:40: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
     template<class V> void accept(V & v) const
                                        ^
C:\dev\boost_1_59_0/boost/bind/bind.hpp:608:28: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
         base_type::accept(v);
                            ^
C:\dev\boost_1_59_0/boost/bind/bind.hpp:680:40: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
     template<class V> void accept(V & v) const
                                        ^
C:\dev\boost_1_59_0/boost/bind/bind.hpp:682:28: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
         base_type::accept(v);
                            ^
C:\dev\boost_1_59_0/boost/bind/bind.hpp:757:40: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
     template<class V> void accept(V & v) const
                                        ^
C:\dev\boost_1_59_0/boost/bind/bind.hpp:759:28: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
         base_type::accept(v);
                            ^
C:\dev\boost_1_59_0/boost/bind/bind.hpp:837:40: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
     template<class V> void accept(V & v) const
                                        ^
C:\dev\boost_1_59_0/boost/bind/bind.hpp:839:28: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
         base_type::accept(v);
                            ^
In file included from C:\dev\boost_1_59_0/boost/bind/bind.hpp:1163:0,
                 from C:\dev\boost_1_59_0/boost/bind.hpp:22,
                 from C:\dev\boost_1_59_0/boost/thread/detail/thread.hpp:30,
                 from C:\dev\boost_1_59_0/boost/thread/thread_only.hpp:22,
                 from C:\dev\boost_1_59_0/boost/thread/thread.hpp:12,
                 from C:\dev\boost_1_59_0/boost/thread.hpp:13,
                 from src/libslic3r/libslic3r.h:10,
                 from src/xsinit.h:53,
                 from src\libslic3r\utils.cpp:1:
C:\dev\boost_1_59_0/boost/bind/bind_template.hpp:326:40: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
     template<class V> void accept(V & v) const
                                        ^
C:\dev\boost_1_59_0/boost/bind/bind_template.hpp:334:20: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
         l_.accept(v);
                    ^
In file included from C:\dev\boost_1_59_0/boost/bind.hpp:22:0,
                 from C:\dev\boost_1_59_0/boost/thread/detail/thread.hpp:30,
                 from C:\dev\boost_1_59_0/boost/thread/thread_only.hpp:22,
                 from C:\dev\boost_1_59_0/boost/thread/thread.hpp:12,
                 from C:\dev\boost_1_59_0/boost/thread.hpp:13,
                 from src/libslic3r/libslic3r.h:10,
                 from src/xsinit.h:53,
                 from src\libslic3r\utils.cpp:1:
C:\dev\boost_1_59_0/boost/bind/bind.hpp:1538:17: error: macro "PerlSock_accept" requires 3 arguments, but only 1 given
     t.accept( v );
                 ^
In file included from C:/Strawberry/c/x86_64-w64-mingw32/include/c++/locale:41:0,
                 from C:\dev\boost_1_59_0/boost/io/ios_state.hpp:17,
                 from C:\dev\boost_1_59_0/boost/thread/detail/thread.hpp:35,
                 from C:\dev\boost_1_59_0/boost/thread/thread_only.hpp:22,
                 from C:\dev\boost_1_59_0/boost/thread/thread.hpp:12,
                 from C:\dev\boost_1_59_0/boost/thread.hpp:13,
                 from src/libslic3r/libslic3r.h:10,
                 from src/xsinit.h:53,
                 from src\libslic3r\utils.cpp:1:
C:/Strawberry/c/x86_64-w64-mingw32/include/c++/bits/locale_facets_nonio.h:1768:65: error: macro "PerlLIO_open" passed 3 arguments, but takes just 2
       open(const basic_string<char>&, const locale&, const char*) const;
                                                                 ^
In file included from C:/Strawberry/c/x86_64-w64-mingw32/include/c++/bits/locale_facets_nonio.h:1898:0,
                 from C:/Strawberry/c/x86_64-w64-mingw32/include/c++/locale:41,
                 from C:\dev\boost_1_59_0/boost/io/ios_state.hpp:17,
                 from C:\dev\boost_1_59_0/boost/thread/detail/thread.hpp:35,
                 from C:\dev\boost_1_59_0/boost/thread/thread_only.hpp:22,
                 from C:\dev\boost_1_59_0/boost/thread/thread.hpp:12,
                 from C:\dev\boost_1_59_0/boost/thread.hpp:13,
                 from src/libslic3r/libslic3r.h:10,
                 from src/xsinit.h:53,
                 from src\libslic3r\utils.cpp:1:
C:/Strawberry/c/x86_64-w64-mingw32/include/c++/x86_64-w64-mingw32/bits/messages_members.h:54:18: error: macro "PerlLIO_open" passed 3 arguments, but takes just 2
       const char*) const
                  ^
In file included from C:\dev\boost_1_59_0/boost/thread/condition_variable.hpp:14:0,
                 from C:\dev\boost_1_59_0/boost/thread/v2/thread.hpp:14,
                 from C:\dev\boost_1_59_0/boost/thread/thread_only.hpp:26,
                 from C:\dev\boost_1_59_0/boost/thread/thread.hpp:12,
                 from C:\dev\boost_1_59_0/boost/thread.hpp:13,
                 from src/libslic3r/libslic3r.h:10,
                 from src/xsinit.h:53,
                 from src\libslic3r\utils.cpp:1:
C:\dev\boost_1_59_0/boost/thread/win32/condition_variable.hpp:321:60: error: macro "PerlProc_wait" passed 2 arguments, but takes just 1
         void wait(unique_lock<mutex>& m,predicate_type pred)
                                                            ^
C:\dev\boost_1_59_0/boost/thread/win32/condition_variable.hpp:447:51: error: macro "PerlProc_wait" passed 2 arguments, but takes just 1
         void wait(lock_type& m,predicate_type pred)
                                                   ^
In file included from C:\dev\boost_1_59_0/boost/thread.hpp:24:0,
                 from src/libslic3r/libslic3r.h:10,
                 from src/xsinit.h:53,
                 from src\libslic3r\utils.cpp:1:
C:\dev\boost_1_59_0/boost/thread/future.hpp:313:88: error: macro "PerlProc_wait" passed 2 arguments, but takes just 1
             virtual void wait(boost::unique_lock<boost::mutex>& lock, bool rethrow=true)
                                                                                        ^
C:\dev\boost_1_59_0/boost/thread/future.hpp:321:35: error: macro "PerlProc_wait" passed 2 arguments, but takes just 1
                 wait(lock, rethrow);
                                   ^
C:\dev\boost_1_59_0/boost/thread/future.hpp:757:33: error: macro "PerlProc_wait" passed 2 arguments, but takes just 1
             this->wait(lk, false);
                                 ^
C:\dev\boost_1_59_0/boost/thread/future.hpp:765:79: error: macro "PerlProc_wait" passed 2 arguments, but takes just 1
           virtual void wait(boost::unique_lock<boost::mutex>& lk, bool rethrow)
                                                                               ^
C:\dev\boost_1_59_0/boost/thread/future.hpp:771:48: error: macro "PerlProc_wait" passed 2 arguments, but takes just 1
               this->base_type::wait(lk, rethrow);
                                                ^
C:\dev\boost_1_59_0/boost/thread/future.hpp:1617:42: error: macro "PerlProc_wait" passed 2 arguments, but takes just 1
             this->future_->wait(lk, false);
                                          ^
C:\dev\boost_1_59_0/boost/thread/future.hpp:1643:42: error: macro "PerlProc_wait" passed 2 arguments, but takes just 1
             this->future_->wait(lk, false);
                                          ^
C:\dev\boost_1_59_0/boost/thread/future.hpp:1865:46: error: macro "PerlProc_wait" passed 2 arguments, but takes just 1
                 this->future_->wait(lk, false);
                                              ^
C:\dev\boost_1_59_0/boost/thread/future.hpp:1884:46: error: macro "PerlProc_wait" passed 2 arguments, but takes just 1
                 this->future_->wait(lk, false);
                                              ^
In file included from src/xsinit.h:38:0,
                 from src\libslic3r\utils.cpp:1:
C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*' token
 #    define PERL_GET_THX  ((PerlInterpreter *)PERL_GET_CONTEXT)
                                             ^
C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*' token
C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*' token
C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*' token
C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*' token
 #    define PERL_GET_THX  ((PerlInterpreter *)PERL_GET_CONTEXT)
                                             ^
C:\STRAWB~1\perl\lib\CORE/XSUB.h:492:17: note: in expansion of macro 'PERL_GET_THX'
 #  define aTHX  PERL_GET_THX
                 ^
C:\STRAWB~1\perl\lib\CORE/embedvar.h:38:18: note: in expansion of macro 'aTHX'
 #    define vTHX aTHX
                  ^
C:\STRAWB~1\perl\lib\CORE/embedvar.h:66:20: note: in expansion of macro 'vTHX'
 #define PL_Proc   (vTHX->IProc)
                    ^
C:\STRAWB~1\perl\lib\CORE/iperlsys.h:1060:4: note: in expansion of macro 'PL_Proc'
  (*PL_Proc->pSleep)(PL_Proc, (t))
    ^
C:\STRAWB~1\perl\lib\CORE/XSUB.h:647:20: note: in expansion of macro 'PerlProc_sleep'
 #    define sleep  PerlProc_sleep
                    ^
C:\dev\boost_1_59_0/boost/thread/win32/thread_primitives.hpp:415:25: note: in expansion of macro 'sleep'
             inline void sleep(unsigned long milliseconds)
                         ^
C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*' token
 #    define PERL_GET_THX  ((PerlInterpreter *)PERL_GET_CONTEXT)
                                             ^
C:\STRAWB~1\perl\lib\CORE/XSUB.h:492:17: note: in expansion of macro 'PERL_GET_THX'
 #  define aTHX  PERL_GET_THX
                 ^
C:\STRAWB~1\perl\lib\CORE/embedvar.h:38:18: note: in expansion of macro 'aTHX'
 #    define vTHX aTHX
                  ^
C:\STRAWB~1\perl\lib\CORE/embedvar.h:66:20: note: in expansion of macro 'vTHX'
 #define PL_Proc   (vTHX->IProc)
                    ^
C:\STRAWB~1\perl\lib\CORE/iperlsys.h:1060:4: note: in expansion of macro 'PL_Proc'
  (*PL_Proc->pSleep)(PL_Proc, (t))
    ^
C:\STRAWB~1\perl\lib\CORE/XSUB.h:647:20: note: in expansion of macro 'PerlProc_sleep'
 #    define sleep  PerlProc_sleep
                    ^
C:\dev\boost_1_59_0/boost/thread/win32/thread_primitives.hpp:415:25: note: in expansion of macro 'sleep'
             inline void sleep(unsigned long milliseconds)
                         ^
C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*' token
 #    define PERL_GET_THX  ((PerlInterpreter *)PERL_GET_CONTEXT)
                                             ^
C:\STRAWB~1\perl\lib\CORE/XSUB.h:492:17: note: in expansion of macro 'PERL_GET_THX'
 #  define aTHX  PERL_GET_THX
                 ^
C:\STRAWB~1\perl\lib\CORE/embedvar.h:38:18: note: in expansion of macro 'aTHX'
 #    define vTHX aTHX
                  ^
C:\STRAWB~1\perl\lib\CORE/embedvar.h:66:20: note: in expansion of macro 'vTHX'
 #define PL_Proc   (vTHX->IProc)
                    ^
C:\STRAWB~1\perl\lib\CORE/iperlsys.h:1060:4: note: in expansion of macro 'PL_Proc'
  (*PL_Proc->pSleep)(PL_Proc, (t))
    ^
C:\STRAWB~1\perl\lib\CORE/XSUB.h:647:20: note: in expansion of macro 'PerlProc_sleep'
 #    define sleep  PerlProc_sleep
                    ^
C:\dev\boost_1_59_0/boost/thread/win32/thread_primitives.hpp:415:25: note: in expansion of macro 'sleep'
             inline void sleep(unsigned long milliseconds)
                         ^
C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*' token
 #    define PERL_GET_THX  ((PerlInterpreter *)PERL_GET_CONTEXT)
                                             ^
C:\STRAWB~1\perl\lib\CORE/XSUB.h:492:17: note: in expansion of macro 'PERL_GET_THX'
 #  define aTHX  PERL_GET_THX
                 ^
C:\STRAWB~1\perl\lib\CORE/embedvar.h:38:18: note: in expansion of macro 'aTHX'
 #    define vTHX aTHX
                  ^
C:\STRAWB~1\perl\lib\CORE/embedvar.h:66:20: note: in expansion of macro 'vTHX'
 #define PL_Proc   (vTHX->IProc)
                    ^
C:\STRAWB~1\perl\lib\CORE/iperlsys.h:1060:4: note: in expansion of macro 'PL_Proc'
  (*PL_Proc->pSleep)(PL_Proc, (t))
    ^
C:\STRAWB~1\perl\lib\CORE/XSUB.h:647:20: note: in expansion of macro 'PerlProc_sleep'
 #    define sleep  PerlProc_sleep
                    ^
C:\dev\boost_1_59_0/boost/thread/win32/thread_primitives.hpp:415:25: note: in expansion of macro 'sleep'
             inline void sleep(unsigned long milliseconds)
                         ^
In file included from C:\dev\boost_1_59_0/boost/tuple/tuple.hpp:28:0,
                 from C:\dev\boost_1_59_0/boost/thread/csbl/tuple.hpp:15,
                 from C:\dev\boost_1_59_0/boost/thread/detail/invoker.hpp:32,
                 from C:\dev\boost_1_59_0/boost/thread/future.hpp:20,
                 from C:\dev\boost_1_59_0/boost/thread.hpp:24,
                 from src/libslic3r/libslic3r.h:10,
                 from src/xsinit.h:53,
                 from src\libslic3r\utils.cpp:1:
C:\dev\boost_1_59_0/boost/tuple/detail/tuple_basic.hpp:45:28: error: expected '}' before end of line
 #pragma GCC diagnostic push
                            ^
C:\dev\boost_1_59_0/boost/tuple/detail/tuple_basic.hpp:45:28: error: expected '}' before end of line
C:\dev\boost_1_59_0/boost/tuple/detail/tuple_basic.hpp:45:28: error: expected '}' before end of line
C:\dev\boost_1_59_0/boost/tuple/detail/tuple_basic.hpp:45:28: error: expected declaration before end of line
error building xs.dll file from 'src\libslic3r\utils.cpp' at C:/Strawberry/perl/lib/ExtUtils/CBuilder/Platform/Windows.pm line 129.
FAIL
! Installing ./xs failed. See C:\Users\Martin\.cpanm\work\1480216552.10848\build.log for details. Retry with --force to force install it.
The XS/C++ code failed to compile, aborting

You found more Perl macros that probably need undefining.

"Nothing unreal exists." - Kiri-kin-tha's First Law of Metaphysics.

On Sat, Nov 26, 2016 at 9:27 PM, Tinchus2009 <notifications@github.com>
wrote:

> I just have issued a GIT PULL command, now there are errors reported in
> the build:
>
> C:\dev\boost_1_59_0/boost/bind/bind.hpp:467:40: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> template void accept(V & v) const
> ^
> C:\dev\boost_1_59_0/boost/bind/bind.hpp:469:28: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> base_type::accept(v);
> ^
> C:\dev\boost_1_59_0/boost/bind/bind.hpp:535:40: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> template void accept(V & v) const
> ^
> C:\dev\boost_1_59_0/boost/bind/bind.hpp:537:28: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> base_type::accept(v);
> ^
> C:\dev\boost_1_59_0/boost/bind/bind.hpp:606:40: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> template void accept(V & v) const
> ^
> C:\dev\boost_1_59_0/boost/bind/bind.hpp:608:28: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> base_type::accept(v);
> ^
> C:\dev\boost_1_59_0/boost/bind/bind.hpp:680:40: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> template void accept(V & v) const
> ^
> C:\dev\boost_1_59_0/boost/bind/bind.hpp:682:28: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> base_type::accept(v);
> ^
> C:\dev\boost_1_59_0/boost/bind/bind.hpp:757:40: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> template void accept(V & v) const
> ^
> C:\dev\boost_1_59_0/boost/bind/bind.hpp:759:28: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> base_type::accept(v);
> ^
> C:\dev\boost_1_59_0/boost/bind/bind.hpp:837:40: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> template void accept(V & v) const
> ^
> C:\dev\boost_1_59_0/boost/bind/bind.hpp:839:28: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> base_type::accept(v);
> ^
> In file included from C:\dev\boost_1_59_0/boost/bind/bind.hpp:1163:0,
> from C:\dev\boost_1_59_0/boost/bind.hpp:22,
> from C:\dev\boost_1_59_0/boost/thread/detail/thread.hpp:30,
> from C:\dev\boost_1_59_0/boost/thread/thread_only.hpp:22,
> from C:\dev\boost_1_59_0/boost/thread/thread.hpp:12,
> from C:\dev\boost_1_59_0/boost/thread.hpp:13,
> from src/libslic3r/libslic3r.h:10,
> from src/xsinit.h:53,
> from src\libslic3r\utils.cpp:1:
> C:\dev\boost_1_59_0/boost/bind/bind_template.hpp:326:40: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> template void accept(V & v) const
> ^
> C:\dev\boost_1_59_0/boost/bind/bind_template.hpp:334:20: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> l_.accept(v);
> ^
> In file included from C:\dev\boost_1_59_0/boost/bind.hpp:22:0,
> from C:\dev\boost_1_59_0/boost/thread/detail/thread.hpp:30,
> from C:\dev\boost_1_59_0/boost/thread/thread_only.hpp:22,
> from C:\dev\boost_1_59_0/boost/thread/thread.hpp:12,
> from C:\dev\boost_1_59_0/boost/thread.hpp:13,
> from src/libslic3r/libslic3r.h:10,
> from src/xsinit.h:53,
> from src\libslic3r\utils.cpp:1:
> C:\dev\boost_1_59_0/boost/bind/bind.hpp:1538:17: error: macro
> "PerlSock_accept" requires 3 arguments, but only 1 given
> t.accept( v );
> ^
> In file included from C:/Strawberry/c/x86_64-w64-
> mingw32/include/c++/locale:41:0,
> from C:\dev\boost_1_59_0/boost/io/ios_state.hpp:17,
> from C:\dev\boost_1_59_0/boost/thread/detail/thread.hpp:35,
> from C:\dev\boost_1_59_0/boost/thread/thread_only.hpp:22,
> from C:\dev\boost_1_59_0/boost/thread/thread.hpp:12,
> from C:\dev\boost_1_59_0/boost/thread.hpp:13,
> from src/libslic3r/libslic3r.h:10,
> from src/xsinit.h:53,
> from src\libslic3r\utils.cpp:1:
> C:/Strawberry/c/x86_64-w64-mingw32/include/c++/bits/locale_facets_nonio.h:1768:65:
> error: macro "PerlLIO_open" passed 3 arguments, but takes just 2
> open(const basic_string&, const locale&, const char*) const;
> ^
> In file included from C:/Strawberry/c/x86_64-w64-mingw32/include/c++/bits/
> locale_facets_nonio.h:1898:0,
> from C:/Strawberry/c/x86_64-w64-mingw32/include/c++/locale:41,
> from C:\dev\boost_1_59_0/boost/io/ios_state.hpp:17,
> from C:\dev\boost_1_59_0/boost/thread/detail/thread.hpp:35,
> from C:\dev\boost_1_59_0/boost/thread/thread_only.hpp:22,
> from C:\dev\boost_1_59_0/boost/thread/thread.hpp:12,
> from C:\dev\boost_1_59_0/boost/thread.hpp:13,
> from src/libslic3r/libslic3r.h:10,
> from src/xsinit.h:53,
> from src\libslic3r\utils.cpp:1:
> C:/Strawberry/c/x86_64-w64-mingw32/include/c++/x86_64-
> w64-mingw32/bits/messages_members.h:54:18: error: macro "PerlLIO_open"
> passed 3 arguments, but takes just 2
> const char*) const
> ^
> In file included from C:\dev\boost_1_59_0/boost/
> thread/condition_variable.hpp:14:0,
> from C:\dev\boost_1_59_0/boost/thread/v2/thread.hpp:14,
> from C:\dev\boost_1_59_0/boost/thread/thread_only.hpp:26,
> from C:\dev\boost_1_59_0/boost/thread/thread.hpp:12,
> from C:\dev\boost_1_59_0/boost/thread.hpp:13,
> from src/libslic3r/libslic3r.h:10,
> from src/xsinit.h:53,
> from src\libslic3r\utils.cpp:1:
> C:\dev\boost_1_59_0/boost/thread/win32/condition_variable.hpp:321:60:
> error: macro "PerlProc_wait" passed 2 arguments, but takes just 1
> void wait(unique_lock& m,predicate_type pred)
> ^
> C:\dev\boost_1_59_0/boost/thread/win32/condition_variable.hpp:447:51:
> error: macro "PerlProc_wait" passed 2 arguments, but takes just 1
> void wait(lock_type& m,predicate_type pred)
> ^
> In file included from C:\dev\boost_1_59_0/boost/thread.hpp:24:0,
> from src/libslic3r/libslic3r.h:10,
> from src/xsinit.h:53,
> from src\libslic3r\utils.cpp:1:
> C:\dev\boost_1_59_0/boost/thread/future.hpp:313:88: error: macro
> "PerlProc_wait" passed 2 arguments, but takes just 1
> virtual void wait(boost::unique_lockboost::mutex& lock, bool rethrow=true)
> ^
> C:\dev\boost_1_59_0/boost/thread/future.hpp:321:35: error: macro
> "PerlProc_wait" passed 2 arguments, but takes just 1
> wait(lock, rethrow);
> ^
> C:\dev\boost_1_59_0/boost/thread/future.hpp:757:33: error: macro
> "PerlProc_wait" passed 2 arguments, but takes just 1
> this->wait(lk, false);
> ^
> C:\dev\boost_1_59_0/boost/thread/future.hpp:765:79: error: macro
> "PerlProc_wait" passed 2 arguments, but takes just 1
> virtual void wait(boost::unique_lockboost::mutex& lk, bool rethrow)
> ^
> C:\dev\boost_1_59_0/boost/thread/future.hpp:771:48: error: macro
> "PerlProc_wait" passed 2 arguments, but takes just 1
> this->base_type::wait(lk, rethrow);
> ^
> C:\dev\boost_1_59_0/boost/thread/future.hpp:1617:42: error: macro
> "PerlProc_wait" passed 2 arguments, but takes just 1
> this->future_->wait(lk, false);
> ^
> C:\dev\boost_1_59_0/boost/thread/future.hpp:1643:42: error: macro
> "PerlProc_wait" passed 2 arguments, but takes just 1
> this->future_->wait(lk, false);
> ^
> C:\dev\boost_1_59_0/boost/thread/future.hpp:1865:46: error: macro
> "PerlProc_wait" passed 2 arguments, but takes just 1
> this->future_->wait(lk, false);
> ^
> C:\dev\boost_1_59_0/boost/thread/future.hpp:1884:46: error: macro
> "PerlProc_wait" passed 2 arguments, but takes just 1
> this->future_->wait(lk, false);
> ^
> In file included from src/xsinit.h:38:0,
> from src\libslic3r\utils.cpp:1:
> C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*'
> token
> define PERL_GET_THX ((PerlInterpreter *)PERL_GET_CONTEXT)
>
>                                          ^
>
> C:\STRAWB1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*'
> token
> C:\STRAWB1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '
> *' token C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')'
> before '*' token
> C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*'
> token
> define PERL_GET_THX ((PerlInterpreter *)PERL_GET_CONTEXT)
>
>                                          ^
>
> C:\STRAWB~1\perl\lib\CORE/XSUB.h:492:17: note: in expansion of macro
> 'PERL_GET_THX'
> define aTHX PERL_GET_THX
>
>              ^
>
> C:\STRAWB~1\perl\lib\CORE/embedvar.h:38:18: note: in expansion of macro
> 'aTHX'
> define vTHX aTHX
>
>               ^
>
> C:\STRAWB1\perl\lib\CORE/embedvar.h:66:20: note: in expansion of macro
> 'vTHX'
> #define PL_Proc (vTHX->IProc)
> ^
> C:\STRAWB1\perl\lib\CORE/iperlsys.h:1060:4: note: in expansion of macro
> 'PL_Proc'
> (*PL_Proc->pSleep)(PL_Proc, (t))
> ^
> C:\STRAWB~1\perl\lib\CORE/XSUB.h:647:20: note: in expansion of macro
> 'PerlProc_sleep'
> define sleep PerlProc_sleep
>
>                 ^
>
> C:\dev\boost_1_59_0/boost/thread/win32/thread_primitives.hpp:415:25:
> note: in expansion of macro 'sleep'
> inline void sleep(unsigned long milliseconds)
> ^
> C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*'
> token
> define PERL_GET_THX ((PerlInterpreter *)PERL_GET_CONTEXT)
>
>                                          ^
>
> C:\STRAWB~1\perl\lib\CORE/XSUB.h:492:17: note: in expansion of macro
> 'PERL_GET_THX'
> define aTHX PERL_GET_THX
>
>              ^
>
> C:\STRAWB~1\perl\lib\CORE/embedvar.h:38:18: note: in expansion of macro
> 'aTHX'
> define vTHX aTHX
>
>               ^
>
> C:\STRAWB1\perl\lib\CORE/embedvar.h:66:20: note: in expansion of macro
> 'vTHX'
> #define PL_Proc (vTHX->IProc)
> ^
> C:\STRAWB1\perl\lib\CORE/iperlsys.h:1060:4: note: in expansion of macro
> 'PL_Proc'
> (*PL_Proc->pSleep)(PL_Proc, (t))
> ^
> C:\STRAWB~1\perl\lib\CORE/XSUB.h:647:20: note: in expansion of macro
> 'PerlProc_sleep'
> define sleep PerlProc_sleep
>
>                 ^
>
> C:\dev\boost_1_59_0/boost/thread/win32/thread_primitives.hpp:415:25:
> note: in expansion of macro 'sleep'
> inline void sleep(unsigned long milliseconds)
> ^
> C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*'
> token
> define PERL_GET_THX ((PerlInterpreter *)PERL_GET_CONTEXT)
>
>                                          ^
>
> C:\STRAWB~1\perl\lib\CORE/XSUB.h:492:17: note: in expansion of macro
> 'PERL_GET_THX'
> define aTHX PERL_GET_THX
>
>              ^
>
> C:\STRAWB~1\perl\lib\CORE/embedvar.h:38:18: note: in expansion of macro
> 'aTHX'
> define vTHX aTHX
>
>               ^
>
> C:\STRAWB1\perl\lib\CORE/embedvar.h:66:20: note: in expansion of macro
> 'vTHX'
> #define PL_Proc (vTHX->IProc)
> ^
> C:\STRAWB1\perl\lib\CORE/iperlsys.h:1060:4: note: in expansion of macro
> 'PL_Proc'
> (*PL_Proc->pSleep)(PL_Proc, (t))
> ^
> C:\STRAWB~1\perl\lib\CORE/XSUB.h:647:20: note: in expansion of macro
> 'PerlProc_sleep'
> define sleep PerlProc_sleep
>
>                 ^
>
> C:\dev\boost_1_59_0/boost/thread/win32/thread_primitives.hpp:415:25:
> note: in expansion of macro 'sleep'
> inline void sleep(unsigned long milliseconds)
> ^
> C:\STRAWB~1\perl\lib\CORE/perl.h:3387:45: error: expected ')' before '*'
> token
> define PERL_GET_THX ((PerlInterpreter *)PERL_GET_CONTEXT)
>
>                                          ^
>
> C:\STRAWB~1\perl\lib\CORE/XSUB.h:492:17: note: in expansion of macro
> 'PERL_GET_THX'
> define aTHX PERL_GET_THX
>
>              ^
>
> C:\STRAWB~1\perl\lib\CORE/embedvar.h:38:18: note: in expansion of macro
> 'aTHX'
> define vTHX aTHX
>
>               ^
>
> C:\STRAWB1\perl\lib\CORE/embedvar.h:66:20: note: in expansion of macro
> 'vTHX'
> #define PL_Proc (vTHX->IProc)
> ^
> C:\STRAWB1\perl\lib\CORE/iperlsys.h:1060:4: note: in expansion of macro
> 'PL_Proc'
> (*PL_Proc->pSleep)(PL_Proc, (t))
> ^
> C:\STRAWB~1\perl\lib\CORE/XSUB.h:647:20: note: in expansion of macro
> 'PerlProc_sleep'
> define sleep PerlProc_sleep
>
>                 ^
>
> C:\dev\boost_1_59_0/boost/thread/win32/thread_primitives.hpp:415:25:
> note: in expansion of macro 'sleep'
> inline void sleep(unsigned long milliseconds)
> ^
> In file included from C:\dev\boost_1_59_0/boost/tuple/tuple.hpp:28:0,
> from C:\dev\boost_1_59_0/boost/thread/csbl/tuple.hpp:15,
> from C:\dev\boost_1_59_0/boost/thread/detail/invoker.hpp:32,
> from C:\dev\boost_1_59_0/boost/thread/future.hpp:20,
> from C:\dev\boost_1_59_0/boost/thread.hpp:24,
> from src/libslic3r/libslic3r.h:10,
> from src/xsinit.h:53,
> from src\libslic3r\utils.cpp:1:
> C:\dev\boost_1_59_0/boost/tuple/detail/tuple_basic.hpp:45:28: error:
> expected '}' before end of line
> #pragma GCC diagnostic push
> ^
> C:\dev\boost_1_59_0/boost/tuple/detail/tuple_basic.hpp:45:28: error:
> expected '}' before end of line
> C:\dev\boost_1_59_0/boost/tuple/detail/tuple_basic.hpp:45:28: error:
> expected '}' before end of line
> C:\dev\boost_1_59_0/boost/tuple/detail/tuple_basic.hpp:45:28: error:
> expected declaration before end of line
> error building xs.dll file from 'src\libslic3r\utils.cpp' at
> C:/Strawberry/perl/lib/ExtUtils/CBuilder/Platform/Windows.pm line 129.
> FAIL
> ! Installing ./xs failed. See C:\Users\Martin.cpanm\work\1480216552.10848\build.log
> for details. Retry with --force to force install it.
> The XS/C++ code failed to compile, aborting
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/alexrj/Slic3r/issues/3583#issuecomment-263100100>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAB8ClrfzTYhQvVCv5vpeqwxbIYW6q72ks5rCPipgaJpZM4K6Fxp>
> .
>

--- a/xs/src/xsinit.h
+++ b/xs/src/xsinit.h
@@ -31,6 +31,7 @@
 #include <ostream>
 #include <iostream>
 #include <sstream>
+#include <libslic3r.h>
 
 #ifdef SLIC3RXS
 extern "C" {
@@ -50,7 +51,6 @@ extern "C" {
 }
 #endif
 
-#include <libslic3r.h>
 #include <ClipperUtils.hpp>
 #include <Config.hpp>
 #include <ExPolygon.hpp>

